### PR TITLE
feat: i18n copy navigation items [CU-29pdxpx]

### PR DESCRIPTION
**Pull request by CodeVoyager**
## Ticket

[[CU-29pdxpx]](https://app.clickup.com/t/29pdxpx)

## Summary

- copying of navigation items as a mean of bootstrapping navigations locale
- minor typing changes
- minor code refactors

image

## Test Plan

- prepare an empty navigation
- copy from non-empty navigation
- copying should required confirmation from an user
- localised items should copied in fashion that item locale matches target's locale (if locale version present)
- if localised item is not present plugin should fallback to original item
- non-localised content should be copied as-is
- copying to non-empty navigation should be forbidden

### CODE REVIEW

**Comment by cyp3rius** [comment]
Task linked: [CU-29pdxpx i18n support  - implementation](https://app.clickup.com/t/29pdxpx)

**Review by ltsNotMike — APPROVED** [review]
👌

File: server/utils/functions.ts [hunk 1]
```diff
@@ -17,7 +17,9 @@ import { NavigationError } from '../../utils/NavigationError';
 import { TEMPLATE_DEFAULT, ALLOWED_CONTENT_TYPES, RESTRICTED_CONTENT_TYPES } from './constant';
 declare var strapi: IStrapi;
 
-export const getPluginService = <T>(name: string): T =>
+export type ServiceKey = 'common' | 'admin' | 'client';
```
[the remark this answers is not in the document]
Comment (CodeVoyager) [reply server/utils/functions.ts:20]: From what I can see my declaration is redundant since your changes. Will apply changes from `develop`.

**Review by ltsNotMike — APPROVED** [review]

**Review by cyp3rius — COMMENTED** [review]
From code perspective, all perfect :) Got just a question in regard to `copy from` buttons can it be dropdown? and additional action button?

**Comment by CodeVoyager** [comment]
I pushed a new approach. Not super sure about details. I'm open to creative input.

**Pull request merged**